services/controlplane/identity/identity.go

```diff
@@ -1,11 +1,14 @@
 package identity
 
+import "github.com/rudderlabs/rudder-server/utils/types/deployment"
+
 // Identifier abstracts how data-plane can be identified to the control-plane.
 //
 //	Including both a unique identifier and the authentication method.
 type Identifier interface {
 	ID() string
 	BasicAuth() (string, string)
+	Type() deployment.Type
```

Comment:
Why not use reflection to determine the type? 

It is quite an idiomatic way, the only downside is performance.